File: minseok/10828/10828/10828.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;
stack<int> S;

int main() {
    int n;
    cin >> n;

    for (int i = 0; i < n; i++) {
        string command;
        cin >> command;

        if (command == "push") {
            int x; 
            cin >> x;
            S.push(x);
        }

        else if (command == "top") {
            if (S.empty())cout << "-1" << endl;
            else cout << S.top() << "\n";
        }
        else if (command == "pop") {
            if (S.empty())cout << "-1" << endl;
            else {
                cout << S.top() << endl;
                S.pop();
            }
        }
        else if (command == "empty") {
            cout << S.empty() << endl;
        }
        else {
            cout << S.size() << endl;
        }
    }
    return 0;
}

File: yeonghwan/algorithm-boj/sep_w1/10828.py
from sys import stdin
input = stdin.readline

N = int(input())

stack = []

for _ in range(N):
    command = list(map(str, input().split()))

    if command[0] == "push":
        stack.append(command[1])

    elif command[0] == "pop":
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    elif command[0] == "size":
        print(len(stack))
    elif command[0] == "empty":
        print(1) if len(stack) == 0 else print(0)
    elif command[0] == "top":
        print(stack[-1]) if len(stack) != 0 else print(-1)
